diesel_cli/tests/migration_list.rs

```diff
@@ -0,0 +1,113 @@
+
+    assert!(!db.table_exists("users"));
+
+    // finds unapplied migration
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    assert!(result.stdout().contains("[ ] 12345_create_users_table"));
+
+    let result = p.command("migration")
+        .arg("run")
+        .run();
+    assert!(result.is_success());
+    assert!(db.table_exists("users"));
+
+    // finds applied migration
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    assert!(result.stdout().contains("[X] 12345_create_users_table"));
+}
+
+#[test]
+fn migration_list_lists_migrations_ordered_by_timestamp() {
+    let p = project("migration_list_ordered")
+        .folder("migrations")
+        .build();
+
+    p.command("setup").run();
+
+    let tag1 = format!("{}_initial", UTC::now().format(TIMESTAMP_FORMAT));
+    p.create_migration(&tag1, "", "");
+
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    assert!(result.stdout().contains(&format!("[ ] {}", &tag1)));
+
+    sleep(Duration::from_millis(1100));
+
+    let tag2 = format!("{}_alter", UTC::now().format(TIMESTAMP_FORMAT));
+    p.create_migration(&tag2, "", "");
+
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    let pattern = format!(r"(\s|.)*\[ \] {}(\s|.)*\[ \] {}(\s|.)*", &tag1, &tag2);
+    let expected = Regex::new(&pattern).unwrap();
+    assert!(expected.is_match(result.stdout()));
+}
+
+#[test]
+fn migration_list_orders_unknown_timestamps_last() {
+    let p = project("migration_list_custom")
+        .folder("migrations")
+        .build();
+
+    p.command("setup").run();
+
+    let tag1 = format!("{}_migration1", UTC::now().format(TIMESTAMP_FORMAT));
+    p.create_migration(&tag1, "", "");
+
+    let tag4 = "abc_migration4";
+    p.create_migration(&tag4, "", "");
+
+    let tag5 = "zzz_migration5";
+    p.create_migration(&tag5, "", "");
+
+    sleep(Duration::from_millis(1100));
+
+    let tag2 = format!("{}_migration2", UTC::now().format(TIMESTAMP_FORMAT));
+    p.create_migration(&tag2, "", "");
+
+    sleep(Duration::from_millis(1100));
+
+    let tag3 = format!("{}_migration3", UTC::now().format(TIMESTAMP_FORMAT));
+    p.create_migration(&tag3, "", "");
+
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    let pattern = format!(
+        r"(\s|.)*\[ \] {}(\s|.)*\[ \] {}(\s|.)*\[ \] {}(\s|.)*\[ \] {}(\s|.)*\[ \] {}(\s|.)*",
+        &tag1, &tag2, &tag3, &tag4, &tag5);
+    let expected = Regex::new(&pattern).unwrap();
+    assert!(expected.is_match(result.stdout()));
```

Comment:
I left explicitly formatted datetimes out of the expected output in case the `migrations::TIMESTAMP_FORMAT` ever changes.